src/NotepadNext/ScintillaNext.h

```diff
@@ -133,24 +134,44 @@ public slots:
     void renamed();
 
     void lexerChanged();
+    void updateEvalStatusLine(const QString res);
 
 protected:
     void dragEnterEvent(QDragEnterEvent *event) override;
     void dropEvent(QDropEvent *event) override;
+    void keyPressEvent(QKeyEvent *event) override;
 
 private:
     QString name;
     BufferType bufferType = BufferType::New;
     QFileInfo fileInfo;
     QDateTime modifiedTime;
     RangeAllocator indicatorResources;
+    te_parser m_tinyEpr;
```

Comment:
The entire parser should be outside of this class. This tightly couples the parser and all the actions required to the editor itself. There is enough of a public API to this class that it all should be possible to move it out.